pkg/kube/multicluster/secretcontroller.go

```diff
@@ -359,35 +359,37 @@ func (c *Controller) addSecret(name types.NamespacedName, s *corev1.Secret) erro
 			continue
 		}
 
-		action, callback := "Adding", c.handleAdd
+		update := false
 		if prev := c.cs.Get(secretKey, cluster.ID(clusterID)); prev != nil {
-			action, callback = "Updating", c.handleUpdate
 			// clusterID must be unique even across multiple secrets
 			kubeConfigSha := sha256.Sum256(kubeConfig)
 			if bytes.Equal(kubeConfigSha[:], prev.kubeConfigSha[:]) {
-				logger.Infof("skipping update (kubeconfig are identical)")
+				logger.Infof("skipping update (kubeconfigs are identical)")
 				continue
 			}
 			// stop previous remote cluster
 			prev.Stop()
+			c.handleUpdate(prev, prev.stop)
```

Comment:
I think it's a bit confusing that `handleUpdate` AND `handleAdd` now get called for an update.

Reply:
@stevenctl - yes I had the same thinking and its pretty easy to duplicate some of the add code in the update methods to make this cleaner,   but I choose not to do that until there is agreement on the general approach, to minimize the change set people see.